### DISCUSSION TITLE
Goal of the turbo-task crate

### DISCUSSION THREAD

**Discussion by niconicoj** [vercel/turborepo#2370] [Help]
Hi,

Is there any plan to publish the crate `turbo-task` on crates.io in the future, or is it specifically meant for internal use ?

Also, the goals of this crate seems very similar to [salsa](https://github.com/salsa-rs/salsa). What was the rational in making a separate incremental memoization framework ? Are there significant difference between salsa and turbo-task ?

**Comment by jridgewell** [comment]
> Is there any plan to publish the crate turbo-task on crates.io in the future, or is it specifically meant for internal use ?

Yes, we intend to publish all crates.

> Also, the goals of this crate seems very similar to [salsa](https://github.com/salsa-rs/salsa). What was the rational in making a separate incremental memoization framework ? Are there significant difference between salsa and [turbo-task](https://linear.app/vercel/issue/TURBO-task) ?

@padmaia or @sokra could answer this.

**Comment by mcpiroman** [comment]
> @padmaia or @sokra 

Kindly bumping the question.
